### PR TITLE
transport.data.gouv.fr case

### PR DESCRIPTION
**Pull request by josevalim**
Screenshot 2021-11-09 at 22 59 43

---

Screenshot 2021-11-09 at 22-59-03 Embracing open data with Elixir at the Ministry of Ecological Transition in France

### CODE REVIEW

File: _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+As they developed the system, they spotted gaps and errors in the data available. So they began validating the data and reaching out to the institutions publishing them, providing direct feedback and value to the open data platform. The incubator saw the benefits from their contributions and, after a few months, they had a successful state startup in their hands alongside the approval to continue working on their mission.
+
+Between 2017 and 2021, the multi-disciplinary team (one public servant, “business developers”, and technical staff)  worked to increase the coverage of published transportation data and helped cities and operators to reach their technical and regulatory goals.
+
+
+## Current challenges
+
+In 2021, the State Startup has "graduated" from its “beta gouv” incubator and is now part of France's Ministry of Ecological Transition. Now composed by Francis Chabouis,Thibaut Barrère and Antoine Augusti, the technical part of the team is tackling new use cases and challenges as the platform grows in terms of needs and versatility.
```
Comment (AntoineAugusti) [inline _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown:45]:
```suggestion
In 2021, the State Startup has "graduated" from its “beta gouv” incubator and is now part of France's Ministry of Ecological Transition. Now composed by Francis Chabouis, Thibaut Barrère and Antoine Augusti, the technical part of the team is tackling new use cases and challenges as the platform grows in terms of needs and versatility.
```

**Comment by thbar** [comment]
@josevalim the logo is not the one we are authorised to use for this article! We'll come back to you shortly with the validation of that.

File: _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+State startups are a match between a team and a mission. They help "intrapreneurs" - public servants who identify frictions and opportunities to improve the lives of their fellow citizens - tackle real-world challenges alongside a team of experts. This team of 2 to 4 people has six months to build a proof of concept ([see official website](https://beta.gouv.fr/en/)).
+
+The insight is: instead of trying to convince other stakeholders to work towards specific goals, it is better to empower innovation and those who want to drive change. Those individuals are given a budget and the autonomy to choose the technology and assemble their team. In exchange, they must open-source and publicly document all code, costs, metrics, and statistics.
+
+The first State Startup was [data.gouv.fr](https://www.data.gouv.fr/en), which transformed France's open data efforts from a catalog of spreadsheets into a social platform that closed the gap between the citizens consuming the data and the institutions providing them. The tool is fully [open-source](https://github.com/opendatateam/udata), allowing other countries to use it in production too.
```
Comment (fertapric) [inline _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown:25]: Is it "State Startup" a proper noun? If it is, then we should change "State startups" in the paragraphs above, if not, then just write "state startups" :)

File: _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+At the time of writing, [261 State Startups](https://beta.gouv.fr/startups/) have been launched and are in various states of development.
+
+
+## Elixir drives by
+
+In 2017, a team was assembled to begin a new state startup focused on transportation data. A European delegated regulation making it mandatory for institutions and corporations to make transportation data public was in the making. The State startup aimed at preparing the ecosystem actors for this incoming regulatory change.
```
Comment (fertapric) [inline _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown:32]:
```suggestion
In 2017, a team was assembled to begin a new state startup focused on transportation data. An European delegated regulation making it mandatory for institutions and corporations to make transportation data public was in the making. The State startup aimed at preparing the ecosystem actors for this incoming regulatory change.
```

And I would probably rephrase:

> An European delegated regulation making it mandatory for institutions and corporations to make transportation data public was in the making

Maybe something like:

> Institutions and corporations were going to make transportation data public due to an European delegated regulation that was in the making

File: _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+The two developers of the team, Vincent Lara and Mauko Quiroga, had heard about Elixir and were interested in learning more. They understood it could provide a robust but flexible and fun platform to explore the problem space. So [they bootstrapped the application](https://github.com/etalab/transport-site/commit/837a048c37ac31151b51ac09432dbcbff3917de5) with the [Phoenix web framework](https://phoenixframework.org/).
+
+As they developed the system, they spotted gaps and errors in the data available. So they began validating the data and reaching out to the institutions publishing them, providing direct feedback and value to the open data platform. The incubator saw the benefits from their contributions and, after a few months, they had a successful state startup in their hands alongside the approval to continue working on their mission.
+
+Between 2017 and 2021, the multi-disciplinary team (one public servant, “business developers”, and technical staff)  worked to increase the coverage of published transportation data and helped cities and operators to reach their technical and regulatory goals.
```
Comment (fertapric) [inline _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown:40]:
```suggestion
Between 2017 and 2021, the multi-disciplinary team (one public servant, “business developers”, and technical staff) worked to increase the coverage of published transportation data and helped cities and operators to reach their technical and regulatory goals.
```

File: _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+Some of those formats have supporting technologies (validators, converters) written in other languages (Java, Rust, etc), which would be beneficial to integrate with. The team then realized the way forward is to adapt their Elixir system to orchestrate and coordinate those subsystems. Luckily, Elixir has shown to be exceptionally well suited to this task, thanks to the underlying Erlang Virtual Machine, designed for communication systems. Francis Chabouis expands: "We currently need to integrate with internal and external services in a variety of formats. Some are static, some require periodic pulls, and others keep open connections to push data. Elixir allows us to experiment, iterate, and scale those ideas quickly".
+
+Overall, the data catalog now includes:
+
+
+
+* Timetables for buses, subways, and trains, including trips and operators, as real-time updates
```
Comment (fertapric) [inline _posts/2021-11-10-embracing-open-data-with-elixir-at-the-ministry-of-ecological-transition-in-france.markdown:55]: 💅 
```suggestion
Overall, the data catalog now includes:

* Timetables for buses, subways, and trains, including trips and operators, as real-time updates
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**